conftest.py

```diff
@@ -149,92 +145,150 @@ def _send_post(
     return _send_post
 
 
-@pytest.fixture(scope = "session")
+@pytest.fixture(scope="session")
 def make_valid_request(send_post):
-    def _make_valid_request(
-        method: SensorMethod, params: dict | None = None
-    ) -> dict:
+    def _make_valid_request(method: SensorMethod, params: dict | None = None) -> dict:
         payload = make_valid_payload(method=method, params=params)
         sensor_response = send_post(**payload)
-        return sensor_response.get("result", {})
+        return sensor_response
 
     return _make_valid_request
 
 
-@pytest.fixture(scope = "session")
+@pytest.fixture(scope="session")
 def get_sensor_info(make_valid_request):
     def _get_sensor_info():
         log.info("Get sensor info")
         sensor_response = make_valid_request(SensorMethod.GET_INFO)
-        return SensorInfo(**sensor_response)
-    
+        result = None
+
+        if "result" in sensor_response and isinstance(sensor_response["result"], dict):
+            result = SensorInfo(**sensor_response["result"])
+
+        if "error" in sensor_response:
+            result = sensor_response["error"]
+
+        return result
+
     return _get_sensor_info
 
 
 @pytest.fixture(scope="session")
 def get_sensor_reading(make_valid_request):
     def _get_sensor_reading():
         log.info("Get sensor reading")
-        return make_valid_request(SensorMethod.GET_READING)
+        sensor_response = make_valid_request(SensorMethod.GET_READING)
+        result = None
+
+        if isinstance(sensor_response, dict):
+            if "result" in sensor_response:
+                if isinstance(sensor_response["result"], float):
```

Comment:
`isinstance()` check isn't needed here either because that would be assuming too much about the sensor implementation. One day the sensor team may decide that they return readings as integers and this code will break. Here we only care about returning `result` or `error` from the sensor. What to do with the data and whether it's what it's supposed to be we'll worry about in other parts of the code.